apps/champions/lib/champions/battle/simulator.ex

```diff
@@ -268,6 +268,8 @@ defmodule Champions.Battle.Simulator do
   # Calculate the new state of the battle after a step passes for a unit.
   # Updates cooldowns, casts skills and reduces self-affecting modifier durations.
   defp process_step_for_unit(unit, current_state, initial_step_state, history) do
```

Comment:
I think this function is growing a lot and it's starting to be hard to follow it, maybe we can refactor it following the same convention as `simulate_battle`

Reply:
Will do! I've created this issue so we don't lose track of it: https://github.com/lambdaclass/afk_gacha_game/issues/498